src/test/java/org/jabref/logic/util/io/CiteKeyBasedFileFinderTest.java

```diff
@@ -106,4 +103,34 @@ public void findAssociatedFilesInNonExistingDirectoryFindsNothing() throws Excep
         assertEquals(Collections.emptyList(), results);
     }
 
+    @AfterEach
+    void deleteTempFiles() throws IOException{
```

Comment:
Is this really necessary? Usually the `TempDirectory` extension takes care of file deletion.

Reply:
I think `rootDir = temporaryFolder.getParent();` is the reason why those files didn't disappear automatically. 
In fact, I can just use `deleteIfExists(rootDir);` to solve this problem, but the temp folder on my machine has some other stuff that can't be deleted (they share one temp file folder).  So I have to use this to guarantee the tests run in a correct environment.

Reply:
Good observation. In fact, you should never use the parent of the temporary folder, since a user might not even have write access to it. Can you please change the tests (here and below) so that they don't use the parent. Thanks.